### ISSUE TITLE
check-types can't decide on object vs. Object<>

### ISSUE THREAD

**Issue by DamienCassou** [gajus/eslint-plugin-jsdoc#860]
## Expected behavior

I'm trying to update to version 38.1.0 but I have an issue with "object" arguments. I expect one of the following to be correct.

Either the lowercase "object":

```
 * @param {object<string, string>} spec - Foo.
```

but I get

```
  20:0  error  Invalid JSDoc @param "spec" type "object"; prefer: "Object<>"  jsdoc/check-types
```

or the uppercase object:

```
 * @param {Object<string, string>} spec - Foo.
```

but I get

```
  20:0  error  Invalid JSDoc @param "spec" type "Object"; prefer: "object"  jsdoc/check-types
```

## Actual behavior

I'm very happy having to make my jsdoc uniform, but the check-type rule has to decide whether it wants uppercase or lowercase `object`, it can't be both :-).

## ESLint Config

```js
{
  "env": {},
  "globals": {},
  "parser": null,
  "parserOptions": {},
  "plugins": [
    "jsdoc"
  ],
  "rules": {
    "jsdoc/check-types": [
      "error"
    ]
  },
  "settings": {},
  "ignorePatterns": []
}
```

## ESLint sample

```js
/**
 * Does something.
 *
 * @param {object<string,string>} spec - Foo.
 */
function foo(spec) {
    return spec;
}

foo()

```

## Environment

- Node version: 16.14.2
- ESLint version: v8.12.0
- `eslint-plugin-jsdoc` version: 38.1.0

**Comment by brettz9** [comment]
Apologies, hadn't caught that plain jsdoc mode had another check still present insisting on `object` in all cases.

In the new fixed release, it should be upper-case when there is a child type, e.g., `Object<string, SomeType>`, and lower case when by itself (`object`). This will ensure better compatibility with TypeScript if you ever switch to that mode.

**Comment by DamienCassou** [comment]
@brettz9: There is nothing to apologize for. You are doing a very good job. Thank you so much for fixing this issue so quickly!
